### ISSUE TITLE
'uploadFile' event starts uploading only on second emit (version 3.x.x)

### ISSUE THREAD

**Issue by DmitriyOnypko** [jkuri/ngx-uploader#282]
What is the correct way to trigger one file upload?
```
  uploadFile(file: UploadFile) {
    const event: UploadInput = {
      type: 'uploadFile',
      file: file,
      url: 'http://ngx-uploader.com/upload',
      method: 'POST',
    }
    this.uploadInput.emit(event);
  }
```
does not start upload on first call (it does on subsequent calls). 
Also it triggers uploadOutput events per each ngx-uploader output event subscribe fn on page.
I.E. per each (uploadOutput)="onUploadOutput($event)".
On the demo page you'll get 2 type="start" one after another, several "uploading" and 2 "done" after another 'uploadFile' emit with one file.

**Comment by gilsdav** [comment]
Same issue.
I tried to cancel after calling 'uploadAll' but when I understand that functionality doesn't exist, I tried using 'uploadFile' of my filtered files.
Unfortunately, the web-service is only called the **second time** I drag&drop the **same file**.

**Comment by marco10024** [comment]
Same issue.

**Comment by gilsdav** [comment]
@DmitriyOnypko this is corrected on 3.1.1.
Can you try and close the issue ?
